### PR TITLE
Fixed #237 - Do not call Fragment assembler if Knot processing result…

### LINKED ISSUE #237: Form submission failed
Action Knot sets response status code to 301. It means that user should be redirected to new location - we should not futher process this request. Currently Assebler processes request and responds with error.

### PR DESCRIPTION
**Pull request by marcinczeczko**
Fixed #237 - Do not call Fragment assembler if Knot processing finished with response Code other than 200
- Added unit test to cover this bug

### CODE REVIEW

File: knotx-server/src/main/java/io/knotx/server/KnotxAssemblerHandler.java [hunk 1]
```diff
@@ -51,28 +51,34 @@ static KnotxAssemblerHandler create(Vertx vertx, KnotxServerConfiguration config
...
-    assembler.processObservable(knotContext)
-        .doOnNext(this::traceMessage)
-        .subscribe(
-            ctx -> {
-              if (ctx.getClientResponse().getStatusCode() == HttpResponseStatus.OK.code()) {
-                sendResponse(context, ctx);
-              } else {
-                context.fail(ctx.getClientResponse().getStatusCode());
+    if (knotContext.getClientResponse().getStatusCode() != HttpResponseStatus.OK.code()) {
+      HttpServerResponse httpResponse = context.response();
```
Comment (tomaszmichalak) [inline knotx-server/src/main/java/io/knotx/server/KnotxAssemblerHandler.java]: Why not sendResponse() method here? We can slightly modify this method to support both cases.

File: knotx-server/src/main/java/io/knotx/server/KnotxAssemblerHandler.java [hunk 2]
```diff
@@ -87,9 +93,7 @@ private void writeHeaders(final HttpServerResponse response,
                     .getAll(name)
                     .forEach(value -> response.putHeader(name, value))
         );
-  }
 
-  private void writeContentLength(final HttpServerResponse response) {
```
Comment (tomaszmichalak) [inline knotx-server/src/main/java/io/knotx/server/KnotxAssemblerHandler.java]: Nice, this method was not needed.

File: knotx-server/src/main/java/io/knotx/server/KnotxAssemblerHandler.java [hunk 3]
```diff
@@ -51,28 +51,34 @@ static KnotxAssemblerHandler create(Vertx vertx, KnotxServerConfiguration config
...
 
-    assembler.processObservable(knotContext)
-        .doOnNext(this::traceMessage)
-        .subscribe(
-            ctx -> {
-              if (ctx.getClientResponse().getStatusCode() == HttpResponseStatus.OK.code()) {
-                sendResponse(context, ctx);
-              } else {
-                context.fail(ctx.getClientResponse().getStatusCode());
+    if (knotContext.getClientResponse().getStatusCode() != HttpResponseStatus.OK.code()) {
```
Comment (tomaszmichalak) [inline knotx-server/src/main/java/io/knotx/server/KnotxAssemblerHandler.java]: Can we extract this condition to method?

Comment (marcinczeczko) [inline knotx-server/src/main/java/io/knotx/server/KnotxAssemblerHandler.java]: fixed

File: knotx-server/src/main/java/io/knotx/server/KnotxAssemblerHandler.java (hunk 1 above)
Comment (marcinczeczko) [inline knotx-server/src/main/java/io/knotx/server/KnotxAssemblerHandler.java]: fixed

File: knotx-server/src/main/java/io/knotx/server/KnotxAssemblerHandler.java (hunk 2 above)
Comment (marcinczeczko) [inline knotx-server/src/main/java/io/knotx/server/KnotxAssemblerHandler.java]: :)

**Pull request merged**